src/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import cx from '../utils/classnames';
+import { props, t, ReactChildren } from '../utils';
+import FlexView from 'react-flexview';
+
+export type NavBarProps = {
+  /** Left content: It doesn't shrink nor grow */
+  /** Center content. It grows to use all the available space */
+  /** Right content. It doesn't shrink nor grow */
+  /** Max Width. For css `max-width` */
+  content: {
+    left?: any, // TODO: ReactChildren
```

Comment:
`any`? couldn't be a `JSX.Element | string`?
(same for every other `any` below)

Reply:
As far as I know, there is a typings called `ReactChild` that we could use it but we didn't agree yet and we are typing children everywhere with `any` (and add a `TODO`)

```ts
type ReactChild = ReactElement<any> | ReactText;
```

\cc @FrancescoCioria @giogonzo

Reply:
these are not children though, are other correctly typed props, amiright?

Reply:
they were typed at aruntime as `tcomb.ReactChildren anyway

from @types/react/index.d.ts
```ts
    // React Nodes
    // http://facebook.github.io/react/docs/glossary.html
    // ----------------------------------------------------------------------

    type ReactText = string | number;
    type ReactChild = ReactElement<any> | ReactText;

    // Should be Array<ReactNode> but type aliases cannot be recursive
    type ReactFragment = {} | Array<ReactChild | any[] | boolean>;
    type ReactNode = ReactChild | ReactFragment | boolean | null | undefined;

    //
```

I'd take `ReactNode[]`

Reply:
But trying to assign a string to `content.left` would get me 
```
Type 'string' is not assignable to type 'ReactNode[] | undefined'.
```

🤔

Reply:
I think that's why you declared `children` as an *array* of ReactNodes... try to declare it as `ReactNode` an pass an array of children

Reply:
It should be `ReactNode`, not `ReactNode[]`. This is the definition of `ReactNode`:

```ts
type ReactText = string | number;
type ReactChild = ReactElement<any> | ReactText;

// Should be Array<ReactNode> but type aliases cannot be recursive
type ReactFragment = {} | Array<ReactChild | any[] | boolean>;
type ReactNode = ReactChild | ReactFragment | boolean | null | undefined;
```

`ReactNode` already has `any[]` through `ReactFragment`.

(BTW, `t.ReactChildren` accepts a single child too so even if it was a single node, the `[]` should be removed)

Reply:
> It should be ReactNode, not ReactNode[]. This is the definition of ReactNode:

yes my bad

`ReactNode` seems fine in the end, shouldn't be ok to use it in general for `children` or am I missing something?

Reply:
> ReactNode already has any[] through ReactFragment.

oops, actually it's a `Array<any[]>`

so, if you want to allow the users to pass `left: [<div />, 'poverone']` you should do `ReactNode | ReactNode[]`... but... are we sure we should support that weird case?